utils/singleton.py

```diff
@@ -19,4 +19,18 @@ def __call__(cls, *args, **kwargs):
 def destroyAllSingletonInstances():
     global _instances
     _instances = {}
-    
\ No newline at end of file
+
+def noSingletonsAround():

```

Comment:
Two thoughts here:

1) A more flexible API might be to just return the number of singletons present - this has use outside of a test suite, and is more flexible within the test suite (IE, call getInstance(), check count ticks up by one)
2) Alternatively, move the "check all destroyed" logic into a test-specific .py file. For test purposes, I've got no qualms about accessing private members.